### PR TITLE
Update data structures

### PR DESCRIPTION
**Pull request by ostowe**
- Combine all objects we were passing in before into a single store (including config, kss sections in  various formats, and template/data reference)
- Deprecate memstore and switch to using immutable.js

### CODE REVIEW

File: config/webpack.config.js [hunk 1]
```diff
@@ -49,9 +49,7 @@ export const defaultConfig = {
     kssOptions: {
       multiline: true,
       markdown: true,
-      custom: [
-        'states',
-      ]
+      custom: ['data'],
```
Comment (pattiereaves) [inline config/webpack.config.js]: I'm just curious, what does this map to?

File: src/cli/actions.js [hunk 2]
```diff
@@ -16,345 +16,312 @@ import requireTemplates from './require-templates';
...
-        reject();
-        break;
-    }
-  });
+            // Write separate HTML snippet for description
+            newStore = updateSection(section, filepath, isInline, store);
+            requireTemplates(newStore);
+            console.log(chalk.green(`KSS file changed or added ${section.referenceURI}`));
+          } else {
+            console.log(chalk.magenta(`No KSS found in ${filepath}`));
```
Comment (pattiereaves) [inline src/cli/actions.js]: 💜💙💛💖💚

File: config/webpack.config.js (hunk 1 above)
Comment (ostowe) [inline config/webpack.config.js]: This is where you put a reference to the JSON file in the kss, so for example:
```
// Test HB
//
// A partial for testing the handlebars importer.
//
// markup: test-hb.hbs
//
// data: test-hb.json
//
// Styleguide test.hb
```

Comment (pattiereaves) [inline config/webpack.config.js]: oh so that's how you did custom kss! 💃

**Pull request merged**